### PR TITLE
HARVESTER:  feture & bug fix

### PR DESCRIPTION
**Pull request by WuJun2016**
## harvester feature & bug fix
1. harvester upgrade
2. code refactoring

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
As per other reviews i haven't tested the functionality but gone through the code.

There's a couple of inline comments but also...

The namespace filter has moved to the left, it looks this has broken the way it's hidden for the custom harvester namespace filter. The cluster name now doesn't need truncating
  image

File: components/HarvesterUpgradeProgressBarList.vue [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          </template>
+        </Collapse>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style lang="scss" scoped>
+.bar-list {
+  .fload-r {
```
[the remark this answers is not in the document]
Comment (WuJun2016) [reply components/HarvesterUpgradeProgressBarList.vue]: Done.

File: components/nav/Header.vue [hunk 2]
```diff
@@ -249,6 +251,15 @@ export default {
       <TopLevelMenu v-if="isMultiCluster || !isSingleVirtualCluster"></TopLevelMenu>
     </div>
 
+    <div
```
[the remark this answers is not in the document]
Comment (WuJun2016) [reply components/nav/Header.vue]: Done.

**Comment by WuJun2016** [comment]
Hi @richard-cox. PTAL，  thanks.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**